CleverRaven/Cataclysm-DDA
# fixes#54183

#### Summary
Bugfixes "Fix calling .back() on empty string"


#### Purpose of change
Fixes #54183


#### Describe the solution
Its fixes the bug by checking for empty string before trying to call .back()




#### Testing
1. Open the Diary.
2. Press (n) for new page
3. Press 'enter' to Edit the page
4. Press 'enter' multiple time
game should not crash


#### Additional context
